auctions/views.py

```diff
@@ -17,17 +19,59 @@ class BidStatusView(LoginRequiredMixin, TemplateView):
     """
     template_name = "addon/widget.html"
 
-    def get_context_data(self, **kwargs):
-        url = self.request.GET['url']
+    def _get_bid_and_claim(self, url):
         bid = None
         claim = None
         try:
             bid = Bid.objects.get(user=self.request.user, url=url)
             claim = Claim.objects.get(issue=bid.issue)
         except:
             pass
+        return (bid, claim)
+
+    def get_context_data(self, **kwargs):
+        url = self.request.GET['url']
+        bid, claim = self._get_bid_and_claim(url)
         return dict({'bid': bid, 'claim': claim, 'url': url})
 
+    def post(self, *args, **kwargs):
+        """
+        Save changes to bid and get payment for offer
+        """
+        url = self.request.POST['url']
+        new_ask_amount = self.request.POST['ask']
+        new_offer_amount = self.request.POST['offer']
+
+        redirect_response = redirect(
+            "%s?url=%s" % (reverse('bid-status'), url)
+        )
+
+        if not(new_ask_amount) and not(new_offer_amount):
+            return redirect_response
+
+        bid, claim = self._get_bid_and_claim(url)
+
+        if not bid:
+            bid = Bid(user=self.request.user, url=url)
+            bid.save()
+
+        if new_ask_amount:
+            bid.ask = new_ask_amount
+            bid.save()
+
+        if new_offer_amount:
+            new_offer_amount = Decimal(self.request.POST['offer'])
+            if new_offer_amount > bid.offer:
+                new_offer = bid.make_offer(new_offer_amount)
+                if new_offer.request():
+                    messages.success(self.request, "Thanks for the offer!")
+                    bid.offer = new_offer_amount
+                    bid.save()
```

Comment:
True.  I was avoiding an atomic lock when the Bid and Offer were both changing.  I figure this will get fixed when we move to the celery methods.  

My biggest priority for this change was to get the success or failure to the widget.